controller/admin_controller.php

```diff
@@ -0,0 +1,296 @@
+<?php
+/**
+ *
+ * Ideas extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\ideas\controller;
+
+/**
+ * Admin controller
+ */
+class admin_controller
+{
+	/** @var \phpbb\config\config */
+	protected $config;
+
+	/** @var \phpbb\db\driver\driver_interface */
+	protected $db;
+
+	/** @var \phpbb\language\language */
+	protected $language;
+
+	/** @var \phpbb\log\log */
+	protected $log;
+
+	/** @var \phpbb\request\request */
+	protected $request;
+
+	/** @var \phpbb\template\template */
+	protected $template;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var string */
+	protected $phpbb_root_path;
+
+	/** @var string */
+	protected $php_ext;
+
+	/** @var array */
+	protected $cfg_array = array();
+
+	/** @var array */
+	protected $new_config = array();
+
+	/** @var string */
+	public $u_action;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\config\config              $config               Config object
+	 * @param \phpbb\db\driver\driver_interface $db                   Database object
+	 * @param \phpbb\language\language          $language             Language object
+	 * @param \phpbb\log\log                    $log                  Log object
+	 * @param \phpbb\request\request            $request              Request object
+	 * @param \phpbb\template\template          $template             Template object
+	 * @param \phpbb\user                       $user                 User object
+	 * @param string                            $phpbb_root_path      phpBB root path
+	 * @param string                            $php_ext              php_ext
+	 * @access public
+	 */
+	public function __construct(\phpbb\config\config $config, \phpbb\db\driver\driver_interface $db, \phpbb\language\language $language, \phpbb\log\log $log, \phpbb\request\request $request, \phpbb\template\template $template, \phpbb\user $user, $root_path, $php_ext)
+	{
+		$this->config = $config;
+		$this->db = $db;
+		$this->language = $language;
+		$this->log = $log;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+		$this->phpbb_root_path = $root_path;
+		$this->php_ext = $php_ext;
+	}
+
+	/**
+	 * Display the options a user can configure for this extension
+	 *
+	 * @return null
+	 * @access public
+	 */
+	public function display_options()
+	{
+		// Create a form key for preventing CSRF attacks
+		add_form_key('acp_phpbb_ideas_settings');
+
+		$this->new_config = $this->config;
+		$this->cfg_array = ($this->request->is_set('config')) ? $this->request->variable('config', array('' => ''), true) : $this->new_config;
+
+		// Check if selected user exists
+		$this->check_ideas_topics_poster();
```

Comment:
I do not like this method being called here. Since the function itself is wrapped entirely within an if submit condition, I think it would be better to move this to where we are actually submitting stuff.

So I would move this into the `set_config_options` method, like this:

```php
	public function set_config_options($display_vars = array())
	{
		$submit = $this->request->is_set_post('submit');

		// Check if selected user exists
		if ($submit)
		{		
				$this->check_ideas_topics_poster();
		}

		// We go through the display_vars to make sure no one is trying to set variables he/she is not allowed to
		foreach ($display_vars as $config_name)
```
And then of course you would remove the if submit conditional that is wrapping all the code in the `check_ideas_topics_poster` method.

This way, we're not running extra code that is not needed during the `display_options` method.